### PR TITLE
Feature buttons and inputs behavior

### PR DESCRIPTION
**Pull request by barneaoren**

### CODE REVIEW

**Comment by amir-microsoft** [comment]
Hi @pjirsa,
following your comment, we've fixed the repo to support "deploy to Azure" from any branch, so now you can test this branch by just clicking the button.

File: public/utilities.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            if (!ul) {
+                return;
+            }
+            var untachedButtons = Array.from(ul.getElementsByTagName('button')).filter(function (button) {
+                return !button.classList.contains("touched")
+            });
+            untachedButtons.forEach(function (button) {
+                button.classList.add("touched");
+                button.addEventListener("click", function () {
+                    button.classList.add("button-selected");
```
Comment (rp-dev) [inline public/utilities.js:58]: This should be 'selected-button' (and not _button-selected_) to match what's in style.css

File: public/utilities.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    if (action.type === 'DIRECT_LINE/INCOMING_ACTIVITY') {
+        setTimeout(function () {
+            var ul = document.getElementById("webchat").getElementsByTagName("ul")[1];
+            if (!ul) {
+                return;
+            }
+            var untachedButtons = Array.from(ul.getElementsByTagName('button')).filter(function (button) {
+                return !button.classList.contains("touched")
+            });
+            untachedButtons.forEach(function (button) {
```
Comment (rp-dev) [inline public/utilities.js:55]: typo: untachedButtons

File: public/utilities.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                return !button.hasAttribute("disabled")
+            });
+            activeButtons.forEach(function (button) {
+                button.classList.add("past");
+                button.setAttribute("disabled", true);
+            });
+            var activeInputs = Array.from(ul.getElementsByTagName('input')).filter(function (input) {
+                return !button.hasAttribute("disabled")
+            });
+            activeInputs.forEach(function (button) {
```
Comment (rp-dev) [inline public/utilities.js:96]: should this be function (input) as opposed to _function (button)_

**Comment by amir-microsoft** [comment]
@barneaoren and I spoke and decided that at this point we wish to minimize the scope of this project to a minimum, and any webcaht enhancement will be proposed directly to the WebChat project:
https://github.com/Microsoft/BotFramework-WebChat

**Pull request closed without merging**